### ISSUE TITLE
Ruff format insist on "single line after imports" for pyi files

### ISSUE THREAD

**Issue by kkpattern** [astral-sh/ruff#9353]
When formatting a pyi file, `ruff format` only keeps a single line after the imports. If `ruff.lint.isort.lines-after-imports` is configured to `2`. `ruff format` will conflict with `ruff check`.

This doesn't happen to py files. `ruff format` will keep the two lines after the imports.

Example:

```bash
(.venv) ☁  ruff-demo  cat test.py
from typing import Any


class Foo:
    def bar(self, data: Any) -> None:
        pass
(.venv) ☁  ruff-demo  ruff format --check test.py
1 file already formatted


(.venv) ☁  ruff-demo  cat test.pyi
from typing import Any


class Foo:
    def bar(self, data: Any) -> None: ...
(.venv) ☁  ruff-demo  ruff format --check test.pyi
Would reformat: test.pyi
1 file would be reformatted
(.venv) ☁  ruff-demo
```

Ruff Version: 0.1.9
Ruff Command: `ruff format`

**Comment by charliermarsh** [comment]
This feels like it's "working as intended" even though the option is conflicting. If anything, we may want to warn here, as we do for other incompatible options (`2` is a permitted value, but I don't think we considered that it would conflict for `.pyi` files). The alternative would be to ignore `lines-after-imports` in `.pyi` files.

**Comment by kkpattern** [comment]
Ignore `lines-after-imports` in .pyi files would be nice. Currently, we're using both `ruff check` and `ruff format --check` in our CI pipeline. We do want two lines after the imports in py files. So, in order to make the CI green, we need to skip `ruff check` for all the pyi files for now.

**Comment by MichaReiser** [comment]
Thank you for reporting this issue. That's something I haven't been aware when adding the warnings. 

Warning when setting `lines-after-imports` to a value other than 1 might be noisy, except if we limit it only to warn when formatting a pyi file (which would be different from the other incompatible warnings). It would also not enable the use case mentioned by @kkpattern. 

I think I would start by ignoring `lines-after-imports` in `pyi` files. But we may need to do run a GitHub code search first to find projects that use `lines-after-imports` with a value other than 1 and have `pyi` files to not break their setup.

**Comment by MichaReiser** [comment]
`isort` fixes `lines-after-imports` and `lines-before-imports` to 1 when using the `black` profile

https://github.com/PyCQA/isort/blob/eed5d059a8ea5442266be892322f8815074800a5/isort/output.py#L211-L225
